libraries/AP_InertialSensor/AP_InertialSensor.cpp

```diff
@@ -1401,25 +1401,58 @@ void AP_InertialSensor::wait_for_sample(void)
 
 check_sample:
     if (!_hil_mode) {
-        // we also wait for at least one backend to have a sample of both
-        // accel and gyro. This normally completes immediately.
-        bool gyro_available = false;
-        bool accel_available = false;
+        // now we wait until we have the gyro and accel samples we need
+        uint8_t gyro_available = 0;
```

Comment:
suggest renaming these to to gyro_available_mask and accel_available_mask

Reply:
fixed